candle-examples/examples/quantized/main.rs

```diff
@@ -545,6 +549,16 @@ fn main() -> anyhow::Result<()> {
         let eos_token = *tos.tokenizer().get_vocab(true).get(eos_token).unwrap();
         let start_post_prompt = std::time::Instant::now();
         let mut sampled = 0;
+
+        // Start metal capture
+        #[cfg(feature = "metal")]
+        if args.metal_tracing {
+            use candle::Device;
+            if let Device::Metal(metal_device) = device.clone() {
+                metal_device.capture("/tmp/candle.gputrace")?;
```

Comment:
Let's make the `metal-tracing` arg an `Option<String>` so that it specifies the filename for the trace rather than hardcoding it?

Reply:
Agreed. Adding to this, does it make sense to have this as an env var?
Gpu tracing not dependent on what type of model you’re running.

Reply:
In which case we should log that gpu tracing is enabled

Reply:
Would rather keep it as an explicit argument as we try to not rely on env variable (though there might be one remaining), the gist of it is that they are hard to discover whereas arguments are documented. Also this way it doesn't depend on whether the env variable gets changed by the user code calling candle etc.
(and on another note, candle tries actively to be log free outside of the `main.rs`)

Reply:
Right. I was thinking it would only take effect with the `debug` or `release-with-debug` profiles. I imagine no one is doing gpu tracing in a production environment.
Debug logging is compiled away in release builds.